### DISCUSSION TITLE
Custom icons for -dmenu rows

### DISCUSSION THREAD

**Discussion by ViniiR** [davatorium/rofi#2044] [Q&A] [answered]
rofi version: 1.7.5

Hi, I have seen in countless discussions and issues that this syntax:
"aap\0icon\x1ffolder\n"
"\x00icon\x1ffile-browser\n"
"\x00image\x1f/absolute/path/img.png\n"
allows us to use icons for option rows specifically in -dmenu mode
while I got the Firefox icon working, I'm more interested in this last one that apparently sources a custom PNG but how exactly does it work? I've tested many ways with absolute paths and etc and I can't get it working 
let's say I have "/home/me/img.png" how can I use it as an icon for a -dmenu mode row?

**Comment by DaveDavenport** [comment, marked as answer]
Absoluate path should work fine:

```bash
echo -en "aap\0icon\x1f/home/qball/test.jpg" | rofi -dmenu  -show-icons
```

**Reply by ViniiR** [reply]
yeah it worked, thanks and for anyone searching for this it does require absolute path and in my above example it would be \x0icon\x1f/home/me/img.png
